tools/toascii/main.cpp

```diff
@@ -137,18 +94,42 @@ int main(int argc, const char** argv)
     }
 
     output_filename += ".txt";
+    auto output_arg = arg_parser.GetArgumentValue(kOutput);
+    if (!output_arg.empty())
+    {
+        output_filename = output_arg;
+    }
 
     gfxrecon::decode::FileProcessor file_processor;
     if (file_processor.Initialize(input_filename))
     {
-        gfxrecon::decode::VulkanDecoder       decoder;
-        gfxrecon::decode::VulkanAsciiConsumer ascii_consumer;
-
-        ascii_consumer.Initialize(output_filename);
-        decoder.AddConsumer(&ascii_consumer);
+        FILE* output_file = nullptr;
+        if (gfxrecon::util::platform::StringCompare(output_filename.c_str(), "stdout") == 0)
+        {
+            output_file = stdout;
+        }
+        else
+        {
+            gfxrecon::util::platform::FileOpen(&output_file, output_filename.c_str(), "w");
+        }
 
-        file_processor.AddDecoder(&decoder);
-        file_processor.ProcessAllFrames();
+        if (output_file)
+        {
+            gfxrecon::decode::VulkanAsciiConsumer ascii_consumer;
+            ascii_consumer.Initialize(output_file);
+            gfxrecon::decode::VulkanDecoder decoder;
+            decoder.AddConsumer(&ascii_consumer);
+            file_processor.AddDecoder(&decoder);
+            file_processor.ProcessAllFrames();
+            if (output_file != stdout)
```

Comment:
Add a call to ascii_consumer.Destroy() here, or the closing `}` will only be printed to stdout, and will be lost otherwise because the file is closed before the AsciiConsumer is destructed.  I tested this by making output with a filename and with "stdout" and diffing the results.